apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -1554,11 +1629,124 @@ where
+                                    ctx.arg_type.inner_named_type().clone(),
+                                    &ctx.selection,
+                                )?
+                                .lazy_map(
+                                    &NamedFragments::default(),
+                                    |selection| {
+                                        if let OpPathElement::InlineFragment(fragment) =
+                                            selection.element()?
+                                        {
+                                            if let Some(CompositeTypeDefinitionPosition::Object(
+                                                obj,
+                                            )) = &fragment.type_condition_position
+                                            {
+                                                if !schema
+                                                    .possible_runtime_types(parent_type.clone())?
+                                                    .contains(obj)
+                                                {
+                                                    return Ok(SelectionMapperReturn::None);
+                                                }
+                                            }
+                                        }
+                                        Ok(SelectionMapperReturn::Selection(selection.clone()))
+                                    },
+                                )?;
+                                // TODO: What happens if we can't convert `e` into an EdgeIndex?
+                                let resolution = condition_resolver.resolve(
+                                    (*e).into().unwrap(),
+                                    context,
+                                    excluded_destinations,
+                                    excluded_conditions,
+                                )?;
+                                // TODO: Is this still needed?
+                                // assert(edge.transition.kind === 'FieldCollection', () => `Expected edge to be a FieldCollection edge, got ${edge.transition.kind}`);
+                                let Some(arg_indices) =
+                                    self.graph.subgraph_to_arg_indices.get(&ctx.subgraph_name)
+                                else {
+                                    internal_error!("TODO")
+                                };
+                                let Some(id) = arg_indices.get(&ctx.argument_coordinate).cloned()
+                                else {
+                                    internal_error!("TODO")
+                                };
+
+                                // TODO: This logic was in the JS code. Does a -1 cost me the
+                                // resolution was unsatisfied?
+                                // if (resolution.cost === -1 || totalCost === -1) {
+                                //  totalCost = -1;
```

Comment:
This check is not needed, correct? We can't have `-1` costs nor total costs
```suggestion
```

Reply:
We were using `resolution.cost` as a proxy for the `ConditionResolution`. Since you're checking that another way, it's fine to remove this.